02_housing/Homelessness_metrics_city.do

```diff
@@ -247,187 +224,119 @@ collapse (sum) *homeless* *black* *hispanic* *other* *white* enrollment , by(yea
+	foreach var in homeless  {
+	replace `var'_count_lb=-1 if `var'_count<=2
+	replace `var'_count_ub=-1 if `var'_count<=2
+	replace `var'_share=-1 if `var'_count<=2
+	replace `var'_quality=-1 if `var'_count<=2
+	replace `var'_count=-1 if `var'_count<=2
+	}
 		
-		replace `var'_count=-1 if `var'_count<=2
-		}
+*merge to crosswalk of places/cities
+	merge 1:1 year state city_name using "intermediate/cityfile.dta"
+	tab year _merge
+		drop if _merge==1 // drop cities that don't merge to the crosswalk
+		keep if year<=2018
+		drop _merge
 		
-*check quality
-sum coverage*, d, if homeless_quality==1
-sum coverage*, d, if homeless_quality==2
-sum coverage*, d, if homeless_quality==3
-
-drop enrollment coverage* enroll_* *_districts_suppress min_* count_supp_* *_missing total_race 
+****************
+*Quality Checks
+****************
 
-order year state city_name *homeless* black* hispanic* other* white*
+*check quality - how good is coverage of homeless students (based on nonsuppresion) by quality variables
+	sum coverage*, d, if homeless_quality==1
+	sum coverage*, d, if homeless_quality==2
+	sum coverage*, d, if homeless_quality==3
 
+drop enrollment coverage* enroll_* *_districts_suppress min_* count_supp_* 
+order year state city_name *homeless* 
 gsort -year state city_name
-
-merge 1:1 year state city_name using "intermediate/cityfile.dta"
-tab year _merge
-	*drop cities that don't merge to the urban-provided file
-	drop if _merge==1
-	keep if year>=2019 & year<=2021 // only updating 2019-2021 years
-	drop _merge 
-
+		
 *summary stats to see possible outliers
-bysort year: sum
-bysort state: sum
-
-tab year // total of 486 cities possible
-tab year if homeless_count==.
-* 2019:56/486 2020:54/486 2021:53/486
-
-*check for race/ethnicity 
-		sum share_race_total, detail
-		sum share_race_total if black_count!=-1, detail
-		gen missing = 0
-		replace missing = 1 if share_race_total==.
-		tab year missing, row
-	
-drop city_name state_name share_race_total missing
+	bysort year: sum // Beaumont TX in 2017 had really high share homeless (Hurricane Harvey)
+	bysort state: sum
 
+bysort year: count // total of 2014-2017:485 2018:486 cities possible
+tab year if homeless_count==. // 2014:62/485 2015:62/485 2016:59/485 2017:60/485 2018:58/486
+
+drop city_name state_name
 order year state place
 gsort -year state place
 
-*data quality check 
-gen check1 = 1 if homeless_count<homeless_count_lb
-gen check2 = 1 if homeless_count>homeless_count_ub
-tab check1, m
-tab check2, m 
-drop check*
+*data quality check - is homeless count ever less than lower bound or higher than upperbound
+	gen check1 = 1 if homeless_count<homeless_count_lb
+	gen check2 = 1 if homeless_count>homeless_count_ub
+	tab check1, m
+	tab check2, m 
+	drop check*
+
+*are all quality flags missing if metric is missing
+tab homeless_quality if homeless_share==.
+tab homeless_quality if homeless_count==.
+
+**************
+*Visual Checks
```

Comment:
Similar comment for visual checks. Useful if you know the topic but otherwise need to be explained what should be checked for in the visual.